src/dataset_miner/data_extractor.py

```
@@ -0,0 +1,217 @@
+            for i, page in enumerate(pdf_reader.pages, 1):
+                try:
+                    page_text = page.extract_text()
+                    text += page_text + "\n\n"
+                    logger.debug(f"Processed page {i}/{total_pages}")
+                except Exception as e:
+                    logger.warning(f"Failed to extract text from page {i}: {str(e)}")
+
+            logger.info(f"Extracted {len(text)} characters from {file_path}")
+            return text.strip()
+    except Exception as e:
+        raise TextExtractionError(f"Failed to extract text from PDF: {str(e)}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)